index.bs

```diff
@@ -514,12 +514,30 @@ To <dfn>handle an incoming message</dfn> given a [=WebSocket connection=]
 
 <div algorithm>
 To <dfn>respond with an error</dfn> given a [=WebSocket connection=]
-|connection| and an [=error code=] |code|:
+|connection|, |command id| and |error code|:
 
- 1. Issue: Form a valid JSON |errorObject| given |code|.
+ 1. If |command id| is not a number, let |command id| be null.
 
- 2. [=Send a WebSocket message=] comprised of |errorObject| over
-    |connection|.
+ 1. Let |error data| be a new map matching the <code>Error</code> production in
+    the [=local end definition=], with the <code>error</code> field set to
+    |error code|, the <code>message</code> field set to an
+    implementation-defined string containing a human-readable definition of the
+    error that occured and the <code>stacktrace</code> field optionally set to
+    an implementation-defined string containing a stack trace report of the
+    active stack frames at the time when the error occurred.
+
+ 1. Set the value of <code>id</code> on |error data| to |command id|.
+
+ 1. Assert: |error data| matches the <code>CommandResponse</code> definition in
+    the [=local end definition=].
+
+ 1. Let |response| be the result of [=serialize JSON to bytes|serializing JSON
+    to bytes=] given |error data|.
+
+ 1. Let |serialized| be the result of [=serialize JSON to
+    bytes|serializing JSON to bytes=] given |response|.
```

Comment:
Does that mean we serialize the `error data` twice?

Reply:
Good catch.